src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -903,7 +903,15 @@ fn traits_implemented_by<'a>(
                 ty
             );
             // Fast path: if this is a primitive simple `==` will work
-            let saw_impl = impl_type == ty;
+            // NOTE: the `match` is necessary; see #92662.
+            let saw_impl = impl_type == ty
+                || match (impl_type.kind(), ty.kind()) {
+                    (ty::Adt(impl_def, _), ty::Adt(ty_def, _)) => {
```

Comment:
```suggestion
                    (ty::Adt(impl_def, _), ty::Adt(ty_def, _)) => {
                        // We intentionally don't compare generics because ...
```

Reply:
@camelid I just pushed a comment, see if that's good enough or if you have feedback